src/context.hpp

```diff
@@ -67,7 +63,7 @@ class context {
         }
         return false;
     }
-    bool should_eval_filters() const
+    bool check_new_filter_targets() const
     {
         for (const auto &[target, str] : ruleset_->filter_addresses) {
```

Comment:
tidy wants `any_of`, as in:

```c++
return std::ranges::any_of(ruleset_->filter_addresses, [this](auto &&target) {
        return store_.is_new_target(std::get<const target_index>(target));
});
```

whether it's more readable is questionable though...

Reply:
I think this didn't compile somewhere (maybe locally in my old laptop, maybe in CI, I can't recall), so I'm ignoring it for now.

Reply:
Okay I tried, let's see if it compiles...

Reply:
Indeed it failed on macos and the old docker builds.